File: DSA/Greedy/Problems/7.Non-overlapping Intervals/sol.js
var eraseOverlapIntervals = function (intervals) {
  intervals.sort((a, b) => a[1] - b[1]);
  let limit = -Infinity,
    count = 0;
  for (let [start, end] of intervals) {
    if (start >= limit) {
      count++;
      limit = end;
    }
  }
  return intervals.length - count;
};

File: DSA/Greedy/Problems/7.Non-overlapping Intervals/sol.py
class Solution:
    def eraseOverlapIntervals(self, intervals):
        intervals.sort(key=lambda x: x[1])
        limit = float('-inf')
        count = 0
        for start, end in intervals:
            if start >= limit:
                count += 1
                limit = end
        return len(intervals) - count
